### PR TITLE
Replace Typeops with Fast_typeops

### PR DESCRIPTION
**Pull request by SkySkimmer**
Fast_typeops and Typeops did the same thing, except Typeops keep building and destructing `unsafe_judgment`s and exposed more functions like `judge_of_foo`. This PR deduplicates the functionality.

### CODE REVIEW

**Comment by SkySkimmer** [comment]
What's a good way to see if this affects performance?
@mattam82

File: plugins/extraction/extraction.ml [hunk 1]
```diff
@@ -258,7 +258,7 @@ let rec extract_type env db j c args =
     | Const (kn,u as c) ->
 	let r = ConstRef kn in
 	let cb = lookup_constant kn env in
-	let typ,_ = Typeops.type_of_constant env c in
+        let typ = Typeops.type_of_constant_in env c in (* FIXME constraints *)
```
Comment (mattam82) [inline plugins/extraction/extraction.ml]: You can remove the FIXME here

File: plugins/funind/recdef.ml [hunk 2]
```diff
@@ -78,8 +78,10 @@ let def_of_const t =
 
 let type_of_const t =
    match (kind_of_term t) with
-    Const sp -> Typeops.type_of_constant (Global.env()) sp
-    |_ -> assert false
+   | Const sp ->
+      (* FIXME discarding universe constraints *)
```
Comment (mattam82) [inline plugins/funind/recdef.ml]: The FIXME should stay if type_of_const can be called on the body of a universe polymorphic function, then it should take an env to do type_of_constant_in in. Probably the FIXME should hence change

**Review by mattam82 — APPROVED** [review]
Indeed constant_type_in should rather be used, it's rare that client code needs the constraints. It should happen only if they generate fresh universes for a polymorphic constant and want to get the instantiated constraints

**Review by mattam82 — APPROVED** [review]
LGTM

**Comment by mattam82** [comment]
For performance measurements you can test HoTT for example on your machine, they have timing scripts. I launched a job on the continuous integration server to see if this breaks anything, you can also spot large changes in performance using it:
https://ci.inria.fr/coq/view/coq-contribs/job/coq-contribs/642/

Can you see it?

**Comment by SkySkimmer** [comment]
I think the HoTT script is having trouble with the new warnings
https://gist.github.com/SkySkimmer/0e9a63aee6970326956ad9db3386995e

The CI failed in the same way as this compile using trunk https://ci.inria.fr/coq/job/bench-math-comp/946/ somewhere in Mathematical Components.
Not sure how to read performance from there.

File: kernel/typeops.mli [hunk 3]
```diff
@@ -101,23 +101,19 @@ val judge_of_case : env -> case_info
   -> unsafe_judgment -> unsafe_judgment -> unsafe_judgment array
     -> unsafe_judgment
 
-(** Typecheck general fixpoint (not checking guard conditions) *)
-val type_fixpoint : env -> Name.t array -> types array
-    -> unsafe_judgment array -> unit
-
-val type_of_constant : env -> pconstant -> types constrained
+(* val type_of_constant : env -> pconstant -> types constrained *)
```
Comment (pedagand) [inline kernel/typeops.mli]: Is this dead code/comment intentional?

**Comment by SkySkimmer** [comment]
It follows the commented `judge_of_inductive_knowing_parameters`. I could also remove it.

**Comment by JasonGross** [comment]
@SkySkimmer 
> I think the HoTT script is having trouble with the new warnings

Should be fixed by https://github.com/HoTT/HoTT/commit/c427e49d674799b681a77cd6d1adabea40ad1a2b which updates the submodule with https://github.com/JasonGross/coq-scripts/commit/70b32dd2da94b945f4e98064f2fbf78ae0755e66

**Comment by JasonGross** [comment]
@SkySkimmer Alternatively, post-process the pretty-printed table with `sed s'/\[[^\]]\]\n//g' (i.e., strip anything matching `[<some stuff>]\n`)

File: plugins/funind/recdef.ml (hunk 2 above)
Comment (SkySkimmer) [inline plugins/funind/recdef.ml]: Broken sentence? Change to what?

Comment (SkySkimmer) [inline plugins/funind/recdef.ml]: I don't know what to do about this.

**Comment by maximedenes** [comment]
Merged, thanks a lot for contributing!

**Pull request merged**